### DISCUSSION TITLE
Shelly restoring original firmware

### DISCUSSION THREAD

**Discussion by juanfal** [arendst/Tasmota#11229] [Q&A] [answered]
I did a Tasmota firmware OTA flashing on a Shelly EM… Ok so far, but, as I don't find great advantage and interface loading is slower, I want to restore the ORIGINAL firmware.  The problem is Tasmota offers file and url loading of new firmwares but in a format (.bin) incompatible with the .zip format.  This zip contains:

```
manifest.json
esp_init_data_default_v08.bin
fs.bin
rboot.bin
shellyem.bin
```

How to get back to the original?
Could any of these files from the shellyEM.zip files make the magic?

Thanks

**Comment by barbudor** [comment, marked as answer]
OTA
https://github.com/yaourdt/tasmota-to-mgos

**Reply by juanfal** [reply]
Thanks, tried, but…
<img width="403" alt="Captura de pantalla 2021-03-05 a las 14 07 31" src="https://user-images.githubusercontent.com/895596/110119553-4501c680-7dbc-11eb-82bb-8ef5c968046e.png">
<img width="443" alt="Captura de pantalla 2021-03-05 a las 14 07 49" src="https://user-images.githubusercontent.com/895596/110119596-53e87900-7dbc-11eb-9a04-5c71649bb430.png">

**Reply by barbudor** [reply]
You haven't read how to

**Comment by juanfal** [comment]
It’s been an odyssey!  My gosh, but finally I managed to get back to the original.  It helped a lot to access through the computer to the ip (I couldn’t find what!) and curl up the firmware after telling tasmota allow any firmware and updating a mongoose bin

A difficult story to tell

Thank you!

--
Juan miPhone
